integration_tests/system_test.go

```diff
@@ -0,0 +1,127 @@
+/*
+ Copyright 2022 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package integration_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/openconfig/lemming/gnmi/oc/ocpath"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+)
+
+// TestHostname verifies that the hostname configuration paths can be read,
+// updated, and deleted.
+//
+// config_path:/system/config/hostname
+// telemetry_path:/system/state/hostname
+func TestHostname(t *testing.T) {
+	testCases := []struct {
+		description string
+		hostname    string
+	}{
+		{"15 Letters", "abcdefghijkmnop"},
+		{"15 Numbers", "123456789012345"},
+		{"Single Character", "x"},
+		{"Periods", "test.name.example"},
+		{"63 Characters", "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"},
+	}
+
+	dut := ondatra.DUT(t, "dut")
+
+	for _, testCase := range testCases {
+		t.Run(testCase.description, func(t *testing.T) {
+			config := ocpath.Root().System().Hostname().Config()
+			state := ocpath.Root().System().Hostname().State()
+
+			gnmi.Replace(t, dut, config, testCase.hostname)
+
+			t.Run("Get Hostname Config", func(t *testing.T) {
```

Comment:
why is this a subtest?

Reply:
I copied this from featureprofiles. This allows for each of these to have a separate PASS/FAIL message on the test output.

Reply:
sgtm